secret/scanner.go

```diff
@@ -39,31 +55,84 @@ type ScanArgs struct {
 
 func (s Scanner) Scan(args ScanArgs) types.Secret {
 	var findings []types.SecretFinding
+	var allRulesResults []RuleResult
+
+	for _, path := range s.AllowList.Paths {
+		if path.MatchString(args.FilePath) {
+			return types.Secret{}
+		}
+	}
+
 	for _, rule := range s.Rules {
+		var ruleResults []RuleResult
 		// Check if the file path should be scanned by this rule
 		if rule.Path != nil && !rule.Path.MatchString(args.FilePath) {
 			continue
 		}
 
+		skipRule := false
+		for _, path := range rule.AllowList.Paths {
+			if path.MatchString(args.FilePath) {
+				skipRule = true
+				break
+			}
+		}
+		if skipRule {
+			continue
+		}
```

Comment:
I think we can define like

```
func shouldSkipPath(filePath string, allowPaths []*regexp.Regexp) bool {
	for _, r := range allowPaths {
		if r.MatchString(filePath) {
			return true
		}
	}
	return false
}
```

 and

```suggestion
		if shouldSkipPath(args.FilePath, rule.AllowList.Paths) {
			continue
		}
```